transport-native-epoll/src/main/c/netty_unix_socket.c

```diff
@@ -647,6 +655,15 @@ static jint netty_unix_socket_isTcpQuickAck(JNIEnv* env, jclass clazz, jint fd)
     }
     return optval;
 }
+
+static jobject netty_channel_unix_socket_peerCredentials(JNIEnv *env, jclass clazz, jint fd) {
+     struct ucred credentials;
+     if(netty_unix_socket_getOption(env,fd, SOL_SOCKET, SO_PEERCRED, &credentials, sizeof (credentials))) {
+         return NULL;
+     }
+     jobject pc = (*env)->NewObject(env, peerCredentialsClass, peerCredentialsMethodId, credentials.pid, credentials.uid, credentials.gid);
```

Comment:
Done, I do that to help debugging and in release mode with any -O2 it will get optimized out, but returned it directly.